### PR TITLE
NSFS | NC |  Removal of NSFS prefix from user facing components - service, cli, configs and events 

### PR DESCRIPTION
**Pull request by romayalon**
### 1.The following PR renames the minimal user facing components having NSFS prefix - 
1. **Add CLI wrapper** - usage change - `node src/cmd/manage_nsfs.js` -> `noobaa-cli`
* add node noobaa-cli commands to /usr/local/bin on RPM  
3. **Service name** - noobaa_nsfs.service -> noobaa.service
4. **Configurations** - 
- NSFS_WHITELIST -> S3_SERVER_IP_WHITELIST
- NSFS_HEALTH_ENDPOINT_RETRY_COUNT -> NC_HEALTH_ENDPOINT_RETRY_COUNT
- NSFS_HEALTH_ENDPOINT_RETRY_DELAY -> NC_HEALTH_ENDPOINT_RETRY_DELAY
5. **Event** - 
- noobaa_nsfs_crashed -> noobaa_s3_crashed
6. **Health errors** - 
- NOOBAA_NSFS_SERVICE_FAILED -> NOOBAA_SERVICE_FAILED
- NSFS_ENDPOINT_FAILED -> NOOBAA_ENDPOINT_FAILED
- NSFS_ENDPOINT_FORK_MISSING -> NOOBAA_ENDPOINT_FORK_MISSING

7. All related **docs/tests** adopted the new namings as well.
9. Env file rename to noobaa
### Issues: Fixed #xxx / Gap #xxx
1. Gap - Rename of internal references of manage_nsfs will be a part of a follow-up PR.
2. move health script under noobaa-cli

### Testing Instructions:
1. Manual Tests in a linux container + RPM deployment - 
```
systemctl enable noobaa.service --now
noobaa-cli account add...
s3 ops using this account - ls, mb, cp etc
node src/cmd/health
```

- [ ] Doc added/updated
- [ ] Tests added

### CODE REVIEW

File: config.js [hunk 1]
```diff
@@ -825,10 +825,10 @@ config.ALLOW_HTTP = false;
 config.BASE_MODE_CONFIG_FILE = 0o600;
 config.BASE_MODE_CONFIG_DIR = 0o700;
 
-config.NSFS_WHITELIST = [];
+config.NC_WHITELIST = [];
```
Comment (guymguym) [inline config.js:828]: this actually should be S3_WHITELIST as we check it for any S3 request, not just NSFS or NC.

File: src/deploy/noobaa-s3.service [hunk 2]
```diff
@@ -8,7 +8,7 @@ RestartSec=2
 User=root
 Group=root
 ExecStartPre=/usr/local/noobaa-core/bin/node /usr/local/noobaa-core/src/upgrade/upgrade_manager.js --nsfs true --upgrade_scripts_dir /usr/local/noobaa-core/src/upgrade/nsfs_upgrade_scripts
-ExecStart=/usr/local/noobaa-core/bin/node /usr/local/noobaa-core/src/cmd/nsfs.js
+ExecStart=/usr/local/noobaa-core/bin/node /usr/local/noobaa-core/src/cmd/noobaa-s3.js
 EnvironmentFile=-/etc/sysconfig/noobaa_nsfs
```
Comment (guymguym) [inline src/deploy/noobaa-s3.service:12]:
```suggestion
EnvironmentFile=-/etc/sysconfig/noobaa-s3
```

File: package.json [hunk 3]
```diff
@@ -29,7 +29,8 @@
     "start": "node --unhandled-rejections=warn src/cmd",
     "cmd": "node --unhandled-rejections=warn src/cmd",
     "api": "node --unhandled-rejections=warn src/cmd/api",
-    "nsfs": "node --unhandled-rejections=warn src/cmd/nsfs",
+    "noobaa-s3": "node --unhandled-rejections=warn src/cmd/noobaa-s3",
+    "noobaa-cli": "node --unhandled-rejections=warn src/cmd/noobaa-cli",
```
[the remark this answers is not in the document]
Comment (romayalon) [reply package.json]: As we discussed will handle the merge of the scripts in the next PR, removed this change for now

File: config.js (hunk 1 above)
Comment (romayalon) [reply config.js]: changed to S3_CLIENT_IP_WHITELIST as we agreed on

File: src/util/http_utils.js [hunk 4]
```diff
@@ -678,8 +678,8 @@ function authorize_session_token(req, options) {
 function validate_nsfs_whitelist(req) {
     // remove prefix for V4 IPs for whitelist validation
     const client_ip = parse_client_ip(req).replace(/^::ffff:/, '');
-    if (config.NSFS_WHITELIST.length === 0) return;
-    for (const whitelist_ip of config.NSFS_WHITELIST) {
+    if (config.S3_CLIENT_IP_WHITELIST.length === 0) return;
+    for (const whitelist_ip of config.S3_CLIENT_IP_WHITELIST) {
```
Comment (guymguym) [inline src/util/http_utils.js:682]: Wait, aren't we doing the wrong thing here? this should actually be S3_SERVER_IP_WHITELIST ...

If I remember correctly, this whitelist is needed to restrict accepting requests to certain host interfaces, by comparing the server IP's.

But our code (from before this PR) seems to be checking the **client** IP (`req.connection.remoteAddress`).

So unless I am missing something, I think this needs fixing and re-validation ASAP.

@naveenpaul1 @romayalon @rkomandu @madhuthorat

Comment (romayalon) [reply src/util/http_utils.js:682]: @guymguym I created a severity 1 issue - https://github.com/noobaa/noobaa-core/issues/7999
will handle it in a seperate PR

**Review by guymguym — APPROVED** [review]

**Pull request merged**